rdr_service/dao/questionnaire_response_dao.py

```diff
@@ -369,15 +373,19 @@ def _validate_link_ids_from_resource_json_group(self, resource, link_ids):
                         logging.error(f'Questionnaire response contains invalid link ID "{link_id}"')
 
     @staticmethod
-    def _get_module_from_questionnaire_history(questionnaire_history):
-        """ Extract the module code value from a questionnaire_history result JSON data (resource field) """
-
+    def _get_module_name(questionnaire_history: QuestionnaireHistory):
+        """ Use the questionnaire_history to determine the module name """
         # Unittest/lower environments may not have expected questionnaire_history content, so allow for missing data
         result = None
-        history_data = json.loads(questionnaire_history.resource) if questionnaire_history.resource else {}
-        if 'group' in history_data.keys() and 'concept' in history_data['group'].keys():
-            concepts = history_data['group']['concept']
-            result = concepts[0]['code'] if len(concepts) else None
+        if isinstance(questionnaire_history, QuestionnaireHistory):
+            concepts = questionnaire_history.concepts
+            if concepts:
+                concept_code = concepts[0].codeId
+                code_obj = CodeDao().get(concept_code)
+                result = code_obj.value if code_obj else None
+        else:
+            logging.debug(f'Unexpected questionnaire_history parameter type {type(questionnaire_history)}')
```

Comment:
```suggestion
            logging.error(f'Unexpected questionnaire_history parameter type {type(questionnaire_history)}')
```

What do you think of bumping this up to the error level? If this method gets reused and gets passed something else that isn't a QuestionnaireHistory then it might be tough to track down the issue if this is only getting flagged at the debug level.

Reply:
I put this in as defensive code in case the unit test / lower environment QuestionnaireHistory data is not fully populated (to avoid triggering repeated error messages for this), but I raised the level in the most recent commit.